### ISSUE TITLE
Copyrights issue

### ISSUE THREAD

**Issue by k0shk0sh** [ThirtyDegreesRay/OpenHub#12]
Hi there, I've seen so far this class
https://github.com/ThirtyDegreesRay/OpenHub/blob/ed0dadeedf68cf9672133bb78d597f64fb70c210/app/src/main/java/com/thirtydegreesray/openhub/mvp/model/NameParser.java which is copy from FastHub repo which is licensed under GPL-3 and I saw that you including your own license to the file which isn't the same license as mine and no mentioning of the author who did that code/file.

It would be awesome if you can fix this.

Thanks.

**Comment by k0shk0sh** [comment]
There are plenty of stuff which are copy from here and there, for example dark theme is exactly like FastHub I'm happy that you like it and include it in your app, however credits must be given to authors, so please do mention that somewhere in your app including a link to the original repo plus in your readme and fix the licensing where it should be exactly like what the original source has.

**Comment by ThirtyDegreesRay** [comment]
@k0shk0sh Hi, I am sorry about your copyrights issue, I'm already add [FastHub](https://github.com/k0shk0sh/FastHub)&[OctoDroid](https://github.com/slapperwan/gh4a) in my readme([Thanks for open source](https://github.com/ThirtyDegreesRay/OpenHub#thanks-for-open-source)), because of it help me so much in design and some technical difficulties。

But I declare that OpenHub is new GitHub app, not a copy of any other github app. Just at the begining of develop openhub, I use a few utils of FastHub, Some pager desgin inspiration from OctoDroid.

Thanks for open source  [FastHub](https://github.com/k0shk0sh/FastHub)&[OctoDroid](https://github.com/slapperwan/gh4a), it's pretty good.

**Comment by yakov116** [comment]
@ThirtyDegreesRay but the license has to be changed too. You copied the code right from FastHub I compared the 2....

**Comment by ThirtyDegreesRay** [comment]
@yakov116 Because it not a completely file copy, just a few utils functions.

**Comment by yakov116** [comment]
@ThirtyDegreesRay um you want me to show you that it is?

I forked your repo and compared the files with fasthub.

The only difference between the files is that you took out the kotlin files and changed the license. 

You want me to show you?

Just curious why don't you join one of the 2 apps to work together instead of copying? We both can use an extra hand and many more people will benefit from your work!

**Comment by ThirtyDegreesRay** [comment]
@yakov116 What compare tool you use, it's true that "Adding an "l" in the file name" copied from FastHub, please ensure it。

**Comment by yakov116** [comment]
@ThirtyDegreesRay um I just forked your repo and then took the files and compared them...

@k0shk0sh can confirm too. Those are his classes his designs. I mean even the string names are the same!! What are you trying to hide?

**Comment by yakov116** [comment]
@ThirtyDegreesRay um I just forked your repo and then took the files and compared them...

@k0shk0sh can confirm too. Those are his classes his designs. I mean even the string names are the same!! What are you trying to hide?

**Comment by ThirtyDegreesRay** [comment]
@yakov116 In your meaning, if there are any name similar with FastHub in OpenHub, it's a copy.

All github app based on github api, there are some named from github api, so it's posibility that two github app has same name in some files .

In your meaning, I copied from github too.[For your string names issue](https://developer.github.com/v3/repos/#list-your-repositories)

**Comment by yakov116** [comment]
@ThirtyDegreesRay um its amazing how we have the same classes word for word! And the our settings look exactly like yours!! And that out code themes are exactly like yours. Not only that but the coding is the same.

Maybe we should change out license to be under your name since it looks like we stile it from you!

Come on! Live up to the truth my friend.

**Comment by dimqua** [comment]
https://github.com/ThirtyDegreesRay/OpenHub/blob/8afb939acb657e489a6f57259bbcf73a2ec93f3f/app/src/main/java/com/thirtydegreesray/openhub/mvp/model/NameParser.java#L27

:rofl:

**Comment by yakov116** [comment]
@dimqua he removed that line already :mask:

**Comment by k0shk0sh** [comment]
I opened this issue not to delete files and whatsoever, but as an open source yourself you should first understand what licensing is, so I'm just asking to follow exact license in both apps. 
So for example FastHub is under GPL-3 which requires you to have the same exact license.
There are rules which should be followed.

Btw: I'm glad that you took FastHub as an inspiration to your repo :)

**Comment by k0shk0sh** [comment]
Take a look at this 

https://github.com/thermatk/FastHub-Libre

See their licensing

**Comment by dimqua** [comment]
According to the author, OpenHub is not a fork, but obviously it is. Doesn't matter how many changes were made.

**Comment by k0shk0sh** [comment]
@dimqua it definitely isn't a fork, but licensing must be respected, I'm not trying to push so hard on this nor make an issue out of it, I just want people to respect open source and by starting with this we might make a difference.

**Comment by ThirtyDegreesRay** [comment]
@k0shk0sh I delete the file beacuse I can write one for my use.

The license I will be repair.

Thanks.

**Comment by dimqua** [comment]
@k0shk0sh it depends on how do you define a fork. I mean, how many lines of the source code should be copied and changed to call it "a fork"? Anyway, if any code from GPL-licensed project were used, the changes should be released under GPL too.

**Comment by ThirtyDegreesRay** [comment]
@k0shk0sh I hava already add your license to file who copied from yours ([see latest commits](https://github.com/ThirtyDegreesRay/OpenHub/commits/master)).

No matter what, I am thanks so much for your open source FastHub, give me a lot of help at begining. And I don't want to infringement, the license is batch add in AndroidStudio, I don't realize that caused copyrights issue.

By the way, OpenHub is open source too, I don't count on earn by it, so I have no idea to infringement, purpose for a better github app.

**PS: Sorry about my smatter about license, I love open source and respect it.**

**Comment by yakov116** [comment]
@ThirtyDegreesRay There is another issue. That is that you changed in the files (that you copied) to your OWN license and REMOVED the one that created it.

Can you explain what open source is? I can tell you what its not.
Its not that you can copy what you want as long as you give a link to its owner. Its in each file that was copied you give credit to its owner.

Please explain how you understand "Open Source"

**Comment by yakov116** [comment]
@ThirtyDegreesRay I'm disappointed you closed this issue without  resolving this.

**Comment by paulcbetts** [comment]
@k0shk0sh You should contact GitHub Support and file a DMCA request on this repo, since it is a clear copyright violation and the author isn't interested in remediating

**Comment by ThirtyDegreesRay** [comment]
@k0shk0sh All the utils code from FastHub have been removed in OpenHub, if you insist on it's a copyright violation, you contact to GitHub.

**OpenHub is my works writed by my hands, nothing to worry about.**

**Comment by k0shk0sh** [comment]
@ThirtyDegreesRay as I said I don't wanna take any action in my previous comment, but you made me take an action and report to GitHub, I'm not asking you anything impossible, change your license to GPL-v3 and include FastHub in everything that includes it.

For example: themes are FastHub so you shouldn't mix up licensing but use the same license.


No hard feelings here, its very simple :).

**Comment by ThirtyDegreesRay** [comment]
@k0shk0sh Your team insist on OpenHub is complete copy of FastHub because of a few code from FastHub, so I removed your code, and then nothing need to explain.

You insist yours, I insist my.

**Comment by moriel5** [comment]
I did not realize that this had happened, having trusted @ThirtyDegreesRay when he said that OpenHub is not based upon any other GitHub client.

@ThirtyDegreesRay I may not have written much code in my life, however I have read about the different licenses (having written and published a script to download and install Chromium-Dev directly from Google's servers, and having licensed it under the AGPL-3 license, which is even stricter then the GPL-3), and I have to say, that @yakov116 and @k0shk0sh are correct, it is not enough to release the code for it to respect the Open Source idea.

And I too have nothing but respect for your code @ThirtyDegreesRay, so as with the rest of the posts here, this is meant to be constructive crititism.

**Comment by k0shk0sh** [comment]
I'm one step away from taking this repo down, if the OP didn't by Friday night change his license and include proper attribution, I'll send GitHub the green line to shut it down.

**Comment by moriel5** [comment]
Can we please wait for @ThirtyDegreesRay  to first read my post?
I have no idea how long we shall have to wait, however I to give him a chance.

So can you wait instead until sunday?
I do not know @ThirtyDegreesRay personally, however I know that I do not always have time to log into GitHub (even from my phone), and sometimes people only have time at the end of the week.

**Comment by k0shk0sh** [comment]
I don't mind waiting, but he seems to yet take this as a threat or something, he is welcome to use everything from FastHub as long as he respects the license, he did remove some files which are from FastHub but for him to literally remove everything from FastHub he has to rewrite the app from zero. Because I can still find alot of files and other stuff that basically belongs to Fasthub. 

Sunday it is as long as he accept this.

**Comment by moriel5** [comment]
That's good to hear.
By the way, I was wondering, how is it that (not including Google's libraries) FastHub is significantly heavier than OpenHub?

One of the reasons I had moved from ForkHub to OpenHub, rather than to FastHub, was because it was closer to ForkHub's weight (even being lighter than ForkHub).

**Comment by ThirtyDegreesRay** [comment]
@moriel5 Thanks for your words and understanding, it make me clear to think about this issue.

**Thanks so much!**

**Comment by ThirtyDegreesRay** [comment]
@k0shk0sh @yakov116 

1. 8 java file
[CodeWebView.java](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/java/com/thirtydegreesray/openhub/ui/widget/webview/CodeWebView.java), [HtmlHelper.java](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/java/com/thirtydegreesray/openhub/ui/widget/webview/HtmlHelper.java), [AppUtils.java](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/java/com/thirtydegreesray/openhub/util/AppUtils.java), [BundleHelper.java](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/java/com/thirtydegreesray/openhub/util/BundleHelper.java), [GitHubHelper.java](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/java/com/thirtydegreesray/openhub/util/GitHubHelper.java), [PrefUtils.java](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/java/com/thirtydegreesray/openhub/util/PrefUtils.java), [ThemeHelper.java](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/java/com/thirtydegreesray/openhub/util/ThemeHelper.java), [ViewUtils.java](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/java/com/thirtydegreesray/openhub/util/ViewUtils.java)

2. 4 xml file
[colors.xml](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/res/values/colors.xml), [dimens.xml](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/res/values/dimens.xml), [theme_dark.xml](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/res/values/theme_dark.xml), [theme_light.xml](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/res/values/theme_light.xml)

The listed 12 files has some code copied from FastHub, the others code of OpenHub was nothing with FastHub, please don't say a lot files cpoied from FastHub, it should be strict!

**Comment by ThirtyDegreesRay** [comment]
@k0shk0sh @yakov116
At the first of @k0shk0sh opened this issue, I was pleasure to resolve it. But @k0shk0sh and @yakov116 push it to that a lot of code copied or basic from FastHub, it make me unacceptable. Just to think about that, it's truely my mistake first to copied some code from FastHub without copyrights, but you absolute say that a lot of code from FastHub, what is "a lot" meaning, it mean most of all? I can't accpet it that my code become yours copy, so I reject to accept your mind.

For examples, @yakov116 said that:"[IActivityContract.java](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/java/com/thirtydegreesray/openhub/mvp/contract/IActivityContract.java), [IAddIssueContract.java](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/java/com/thirtydegreesray/openhub/mvp/contract/IAddIssueContract.java), etc..., was copied from FastHub", it was amazing. Let me tell you, it was a MVP Contract come from google MVP samples.

**At the end, I could change license to GPL-3 and add @k0shk0sh copyright at the listed 12 files. But @k0shk0sh and @yakov116 should't be mix that the other code was from FastHub. That's my principle I insits.**

**Comment by yakov116** [comment]
@ThirtyDegreesRay You want me to show you more? Dont be a fool I can show you left and right!

**Comment by k0shk0sh** [comment]
Hmmm, what about the icons? Aren't they copied from there and has ?icon_color? Don't be so mad, its fine to admit that you got help from the OS community, I got alot of help from there toooo. So calm down and take this very easily.

**Comment by ThirtyDegreesRay** [comment]
@k0shk0sh The icons was created by AndroidStudio.
You say the name of icon_color was yours, okay, add [attrs.xml](https://github.com/ThirtyDegreesRay/OpenHub/blob/v1.6.0/app/src/main/res/values/attrs.xml), and more ?

**Comment by ThirtyDegreesRay** [comment]
@yakov116 Just do it, you be good at compare.

**Comment by k0shk0sh** [comment]
I don't have time to look at everything I said that to prove you a point, you are still taking this so serious and being impolite, why don't you just do what you said and this conversation could end? It's pretty simple. I'm really getting annoyed by replying to the same thing over and over and over.

**Comment by ThirtyDegreesRay** [comment]
@k0shk0sh I hava already change license to GPL-3 and add your copyright at the listed 13 files ([see latest commits](https://github.com/ThirtyDegreesRay/OpenHub/commits/master)).

If there are any other issues, you can tell me.

**Comment by Poussinou** [comment]
@k0shk0sh [So crazy](https://github.com/ThirtyDegreesRay/OpenHub/commit/039c63587b192635832ef91ca2394cee8f31cfca#diff-04c6e90faac2675aa89e2176d2eec7d8) 👍

**Comment by k0shk0sh** [comment]
@Poussinou I have no time to put effort on shutting this down, but you guys cold report it :).

**Comment by yakov116** [comment]
@ThirtyDegreesRay are you crazy?!

**Comment by Poussinou** [comment]
@IzzySoft just to let you know that this developer has some problems with GNU GPL v3. I know you have its application in your repo.

**Comment by IzzySoft** [comment]
@Poussinou can you sum up the current state please? Hard to follow for an outsider. And especially sad as I [have to remove the client @k0shk0sh wrote](https://github.com/k0shk0sh/FastHub/issues/1610) from my repo (he no longer provides `.apk` files).

**Comment by k0shk0sh** [comment]
It isn't bad because most of the code is stolen from FastHub repo whether the opener wants to admit that or not, lets not talk about the theming which they are basically FastHub themes from attributes to everything else. I'm not in the mood yet to take this down but I might consider that.

**Comment by tom-corwin** [comment]
I'm assuming, although closed, this is still an active issue, correct? @k0shk0sh

**Comment by yakov116** [comment]
@tom-corwin yes but I did not follow up with github...

**Comment by tom-corwin** [comment]
Thanks @yakov116, just curious. This issue shouldn't have been closed, or should atleast be reopened, or recreated as a new issue.. This is make or break for some people and it's not like this issue is open, so it's overlooked.

**Comment by dimqua** [comment]
What is the current state?

**Comment by k0shk0sh** [comment]
@dimqua no update so far. But might be as soon as I finish up the new FastHub.

**Comment by 4oo4** [comment]
Wow, I am uninstalling this app immediately @ThirtyDegreesRay seems to have totally failed to grasp why this is an issue, and an extremely tenuous grasp on how copyright and open source licensing both work.

This is extremely FOSS-hostile and I have no faith in the developer anymore. There are tons of other actually FOSS, copyright-respecting GitHub clients out there (will defnitely check out FastHub!), so it's a waste of time to deal with this amateur nonsense.

**Comment by tom-corwin** [comment]
Could anyone compile a list together of what is and isn't copied verbatim from FashHub? Without having to do a deep dive into the files (let alone a language I don't know), it would be nice to see what remains copyrighted to @k0shk0sh in this repository.

**Comment by k0shk0sh** [comment]
@4oo4 the difference is that FastHub-libra is a rip off from FastHub that doesn't contain any google libs such as firebase job dispatcher and analytics. Also it gives everything for free which honestly considered as a hacked application of FastHub. 

@tom-corwin almost everything & I really mean it when I say, almost everything!

**Comment by tom-corwin** [comment]
@k0shk0sh All it would take is a [DCMA Takedown Notice](https://github.com/contact/dmca-notice). I'm tempted to file one myself against this repo, but it's not my work and (legally) I'm not *the copyright owner or authorized to act on the copyright owner’s behalf" (quoted from the DCMA Takedown Notice, as linked to above.

**Comment by moriel5** [comment]
Sorry about my unavailability, I do not have enough time for everything lately.
I personally am also disappointed by @ThirtyDegreesRay's decisions, which is part of why translating GitHub clients has temporarily dropped in importance for me until this issue is resolved.

From a user perspective, I can say that to me, it looks like OpenHub is based upon ForkHub, with elements from FastHub, which makes this even more of a dealbreaker for me, since the only reason why I did not switch to FastHub was because of the weight (I have somewhat of an obsession with ultra-light apps that do not squander functionality, usability, or looks (though that last one is less inportant to me than the other two).
This does raise the question, due to this issue, whether OpenHub also contains code from FastHub, rather than being new, as @ThirtyDegreesRay had claimed.

**Comment by IzzySoft** [comment]
Looks like the story continues with [LMHub](https://github.com/ilokeshmeena/LMHub) (which started with a fresh copy two weeks ago)…

**Comment by k0shk0sh** [comment]
Amazing, another app that wants to provide things that fasthub offer in different ways rather than just contributing to the app. 

Soon this will change hopefully!

**Comment by IzzySoft** [comment]
What always "fascinates" me is how those folks add the entire tree as "initial commit" instead of a fork – and think that way noone would notice…

Forgive my curious question, but what did I miss that will change things soon?

**Comment by k0shk0sh** [comment]
Maybe a new license for the new fasthub with more restrictions and serious rules and regulations. 
But that's an idea that im considering.

**Comment by dimqua** [comment]
@k0shk0sh these people do not care about licenses at all, as long as your project is licensed under any source-available license (even proprietary one) they can violate it and they probably will. :-/

**Comment by WingedDragon** [comment]
7

---Original---

**Comment by IzzySoft** [comment]
If they'd care they'd at least fork and give credit (if only that way) – instead of copying the code pretending they wrote it (and maybe even remove/replace any hints to its origin that might be found inside the code – and yes, I've seen that in the past as well: cloned from one project, global search+replace of all copyright headers, check-in as new project without any other changes, show off how cool software *they* wrote).

While "in dubio pro reo" I like to admit doubt and send a friendly reminder first, with the "full load" of even replacing all pointers to the original dev in the code I feel it unlikely it might have been an "incident", and would even go straight for a take-down. That's the only language I give a chance to being understood.

So as long as your license requires giving credit (which most licenses do), that should be sufficient for this. As @dimqua said: who does copies like that doesn't care for licenses at all (except maybe his own).

**Comment by moriel5** [comment]
Looking at @ThirtyDegreesRay's GitHub history, it seems safe to say that they are no longer working on OpenHub (or anything, from this username), so we have no one to speak with.
@k0shk0sh @IzzySoft, what do you think needs to be done in such a situation?

**Comment by tom-corwin** [comment]
@moriel5 The original code owners (@k0shk0sh, et al) would most likely have to file for a DCMA Takedown (which has beenlinked in a [previous comment](https://github.com/ThirtyDegreesRay/OpenHub/issues/12#issuecomment-455827317)) or contact GitHub by other mean explaining the situation.

**Comment by IzzySoft** [comment]
And only the original code owners can do that (or the takedown request would be rejected).

**Comment by dimqua** [comment]
The development is continuing [in this fork](https://github.com/JoanVC100/OpenHub).
Probably the author (@JoanVC100) could do something to fix legal problems?

**Comment by JoanVC100** [comment]
Okey, let me explain my situation. I was searching for a Github Android Client in Android and I found OpenHub, which was a promising project. After testing the app, I decided that I would like to traduce the app to my native language (Catalan), but I found out that the author stop developing for a few months, so I advised a few contributors to do the pull requests in my fork to continue the development of the UNOFICCIAL app.
Now, I have read all the comments in this issue and I'm aware of the problem. If there is anything I can do for solving the problems with the license of the code, I will do it, even if I have to delete my fork of the code.

**Comment by yakov116** [comment]
@JoanVC100 the best thing you can do is. Join us with building FastHub. 
Currently @k0shk0sh can use some help

**Comment by JoanVC100** [comment]
And what can I do with my fork?

**Comment by tom-corwin** [comment]
@JoanVC100 If you wanted to have your own personal fork, I would probably start by deleting your fork of this repository, and consider using a fresh fork of @k0shk0sh's project, [FastHub](https://github.com/k0shk0sh/FastHub). From there, you may continue to port the code to your preferred language (if that's what you were doing in the first place).

**Comment by JoanVC100** [comment]
No problem. Then, I'm going to delete my fork and start with the Fasthub project. Thanks to @dimqua for the mention in this issue, and thanks @tom-corwin and @yajov116 for the advise and your tips 👍

**Comment by IzzySoft** [comment]
And remember sending in PRs with the changes you made (to the Fasthub repo), so Kosh can merge them and they become part of the project. That way all profit. Thanks a lot for contributing to Fasthub, Joan!

**Comment by JoanVC100** [comment]
Of course, but first I have to prepare the strings.xml file :P

**Comment by IzzySoft** [comment]
@k0shk0sh and should you one day decide you want your app back in my repo, just ring the bell :wink:

**Comment by MohamedElidrissi** [comment]
hey @yakov @k0shk0sh, I'm a fan of FastHub been using it since 2017(writing this comment from it!) I already merged a few commits in @JoanVC100 fork but I think he's deleting it. I would like to contribute to FastHub, since the development on the v5 branch is going slowly I would like if there was some kind of a list with the stuff that you're currently implementing and I can help, even bugs I can squash'em :)

**Comment by daffydock** [comment]
Just read this. Deleting this app immediately.

**Comment by another-sapiens** [comment]
@daffydock nice catch.
